pkg/cloud/aws/services/ec2/gateways.go

```diff
@@ -121,12 +127,37 @@ func (s *Service) createInternetGateway() (*ec2.InternetGateway, error) {
 	}
 
 	s.scope.Info("Created internet gateway for VPC", "vpc-id", s.scope.VPC().ID)
-	_, err = s.scope.EC2.AttachInternetGateway(&ec2.AttachInternetGatewayInput{
+
+	tagParams := s.getGatewayTagParams(*ig.InternetGateway.InternetGatewayId)
+	tagApply := &tags.ApplyParams{
+		EC2Client:   s.scope.EC2,
+		BuildParams: tagParams,
+	}
+	tag := func() (bool, error) {
+		if err := tags.Apply(tagApply); err != nil {
+			return false, err
+		}
+		return true, nil
+	}
+	if err := wait.WaitForWithRetryable(wait.NewBackoff(), tag, awserrors.InternetGatewayNotFound); err != nil {
+		record.Warnf(s.scope.Cluster, "FailedTagInternetGateway", "Failed to tag managed Internet Gateway %q: %v", *ig.InternetGateway.InternetGatewayId, err)
+		return nil, errors.Wrapf(err, "failed to tag internet gateway %q", *ig.InternetGateway.InternetGatewayId)
+	}
+
+	ig.InternetGateway.Tags = converters.MapToTags(v1alpha1.Build(tagParams))
```

Comment:
What's this for?

Reply:
Setting this here makes sure that we are updating the tags after calling ApplyTags, and the later call to EnsureTags is a noop instead of trying to reapply the tags.

Reply:
Could you please add a comment to that effect?